core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -260,6 +315,14 @@ private boolean insertAsSelect(
                     }
                 } catch (Throwable th) {
                     factory = Misc.free(factory);
+                    if (CairoException.isCairoOomError(th) && i < maxRetriesOnOom) {
```

Comment:
```suggestion
                    if (CairoException.isCairoOomError(th) && i < maxRetriesOnOom && intervalStep > 1) {
```

Reply:
Applied in b26181a85d54e49bd37d6fa8da00ca6fe751ac70